Tests/FluentAssertions.Specs/Primitives/ObjectAssertionSpecs.cs

```diff
@@ -189,6 +190,79 @@ public void When_a_value_is_one_of_the_specified_values_it_should_succeed()
             // Assert
             act.Should().NotThrow();
         }
+
+        [Fact]
+        public void When_a_value_is_not_one_of_the_specified_values_and_equality_comparer_passed_it_should_throw()
+        {
+            // Arrange
+            var value = new SomeClass(3);
+
+            // Act
+            Action act = () => value.Should().BeOneOf(new[] { new SomeClass(4), new SomeClass(5) }, new SomeClassEqualityComparer());
+
+            // Assert
+            act
+                .Should().Throw<XunitException>()
+                .WithMessage("Expected value to be one of {SomeClass(4), SomeClass(5)}, but found SomeClass(3).");
+        }
+
+        [Fact]
+        public void When_a_value_is_not_one_of_the_specified_values_and_equality_comparer_passed_it_should_throw_with_descriptive_message()
+        {
+            // Arrange
+            var value = new SomeClass(3);
+
+            // Act
+            Action act = () => value.Should().BeOneOf(new[] { new SomeClass(4), new SomeClass(5) }, "because those are the valid values");
+
+            // Assert
+            act
+                .Should().Throw<XunitException>()
+                .WithMessage("Expected value to be one of {SomeClass(4), SomeClass(5)} because those are the valid values, but found SomeClass(3).");
+        }
```

Comment:
You can merge these two tests, there's no reason to have two tests here as they're testing a single _concept_.
I have written myself a note to clean this up.

The same comment applies for the two tests below.